src/packet.rs

```diff
@@ -508,16 +442,21 @@ packets! {
             0x15 => ClientSettings { locale: String, view_distance: i8, chat_mode: i8, chat_colors: bool, displayed_skin_parts: u8 }
             0x16 => ClientStatus { action_id: Var<i32> }
             0x17 => PluginMessage { channel: String, data: Vec<u8>;
-                impl Packet for PluginMessage {
-                    fn encode(&self, mut dst: &mut Write) -> io::Result<()> {
-                        try!(<String as Protocol>::proto_encode(&self.channel, dst));
-                        try!(dst.write_all(&self.data));
+                impl Protocol for PluginMessage {
+                    type Clean = Self;
+                    fn proto_len(this: &Self) -> usize {
+                        <String as Protocol>::proto_len(&this.channel) + this.data.len()
+                    }
+                    fn proto_encode(this: &Self, mut dst: &mut Write) -> io::Result<()> {
+                        try!(<Var<i32> as Protocol>::proto_encode(&0x17, dst));
```

Comment:
And this as well - both are encoding their IDs - which was necessary in one of the intermediary forms of this commit, but not anymore.